spec/registry.md

```diff
@@ -381,6 +381,69 @@ together with the resolved options' values.
 
 The _function_ `:integer` performs selection as described in [Number Selection](#number-selection) below.
 
+### The `:math` function
+
+The function `:math` is a selector and formatter for matching or formatting
+numeric values to which a mathematical operation has been applied.
+
+> This function is useful for plural selection and formatting of an offset of an input value.
+> For example, it can be used in a message such as this:
+> ```
+> .input {$like_count :integer}
+> .local $others_count = {$like_count :math subtract=1}
+> .match $like_count $others_count
+> 0 * {{Your post has no likes.}}
+> 1 * {{{$name} liked your post.}}
+> * 1 {{{$name} and one other person liked your post.}}
+> * * {{{$name} and {$others_count} other people liked your post.}}
+> ```
+
+#### Operands
+
+The function `:math` requires a [Number Operand](#number-operands) as its _operand_.
+
+#### Options
+
+The options on `:math` are exclusive with each other,
+and exactly one option is always required.
+The options do not have default values.
+
+The following options and their values are
+required in the default registry to be available on the function `:math`:
+- `add`
+  - ([digit size option](#digit-size-options))
+- `subtract`
+  - ([digit size option](#digit-size-options))
+
+If no options or more than one option is set,
+or if an _option_ value is not a [digit size option](#digit-size-options),
+a _Bad Option_ error is emitted
+and a _fallback value_ used as the _resolved value_ of the _expression_.
+
+#### Resolved Value
```

Comment:
Do we need a note about implementation specific boundary conditions, e.g.

```
{$n :math add=$maxIntegerOnMyPlatform}
```

Reply:
I don't think so; that's also covered under _digit size option_:
> Implementations MAY define an upper limit on the _resolved value_ of a digit size option option consistent with that implementation's practical limits.

Reply:
Right, although you can get problems with overflow without approaching the "practical limits" for the option value, e.g.

`{2147483646 :math add=2}  // exceeds Java int by 1`

I think a note is called for. I'll propose one when I do a full review.

Reply:
Tbf, an integer overflow does seem like a practical limit to me.

Reply:
Integer overflow is a "practical limit" but the text only limits the "resolved value of a digit size option", not problems like overflow due to the operation itself. 

I'll suggest some text. It would just be a note, so not material to merging this.